cmd/mailchain/internal/settings/mailbox_state.go

```diff
@@ -56,10 +73,11 @@ type MailboxStateLevelDB struct {
 	Path    values.String
 	Handles values.Int
 	Cache   values.Int
+	kind    values.String
 }
 
 // Produce a leveldb database with settings applied
-func (s MailboxStateLevelDB) Produce() (*ldbstore.Database, error) {
+func (s MailboxStateLevelDB) Produce() (stores.State, error) {
```

Comment:
In Go, functions should accept interfaces and return structs. `MailboxStateLevelDB` is only expected to return a `ldbstore.Database` so interface is not needed